runtime/src/rpc.rs

```diff
@@ -69,6 +74,18 @@ impl InterBtcParachain {
         let api: RuntimeApi = ext_client.clone().to_runtime_api();
 
         let runtime_version = ext_client.rpc().runtime_version(None).await?;
+        if let Some(spec_name) = runtime_version.other.get("specName") {
+            if spec_name == DEFAULT_SPEC_NAME {
+                log::info!("spec_name={}", spec_name);
+            } else {
+                return Err(Error::ParachainMetadataMismatch(
+                    DEFAULT_SPEC_NAME.into(),
+                    // The spec name is always expected to be a string if defined, so `unwrap()` never panics.
+                    spec_name.as_str().unwrap().into(),
```

Comment:
Let's do this and then we can remove the comment above:
```suggestion
                    spec_name.as_str().unwrap_or_default().into(),
```